### ISSUE TITLE
Image pre-trained model for transfer learning

### ISSUE THREAD

**Issue by StuckinPhD** [ludwig-ai/ludwig#746]
Hi,

I have a relatively small image dataset and would like to utilize transfer learning to get better results. I could not find much details related to transfer learning online, and the only mention of it is in the update post here related to BERT: https://eng.uber.com/ludwig-v0-2/.

I am still unsure about how to perform image transfer learning as is possible with using pretrained models on imageNet etc.

Is it possible? if so how would the model definition look for this? as the model starting weights would be different to my reduced classes at output?

Thanks

**Comment by w4nderlust** [comment]
At the moment there areno pretrained image models importable in Ludwig, from version 0.3 on we will start to add them. I'll consider this a feature reqeust.

Also let me know if you would be interested in collaborating in adding this feature, it would be pretty easy.

**Comment by StuckinPhD** [comment]
Id be happy to work on this. 
Ive never contributed to any repo before but let me know what I need to do and I'll do my best :)

**Comment by w4nderlust** [comment]
> Id be happy to work on this.
> how can I help?

Great! Thank you for being willing to help.
We are now orking a lot on porting to TF2, so any contribution should start from the `tf2_porting` branch.
Ludwig has this concept of encoders for input features that makes it easy to add  new pretrained image encoders.
If you look at the `features/image_feature.py` you'll find a disctionary of encoder associated with the input feature.
Ideally you may want to add a pretrained model there, giving it a name and implementing the class. Classes for image encoders are in `model/modules/image_encoders.py`.
Basically one need to implement an object with an `__init__` function that contains the parameters (maybe in this case the path to the pretrained model) and a `call` function that takes as inputs tensors of chape `[batch, h, w, c]` and return either [batch, h', w', c']` or just a `[batch, hidden]` tensord, depending where you cut the original preprocessed model, if before or after flattening.
Let me know if these instructions are clear for you or if you have additional questions.

**Comment by rajatsahay** [comment]
Hey there!
Is anybody still working on this? I would love to help.

**Comment by w4nderlust** [comment]
@rajatsahay thank you for being willing to help! The instructions in my previous message, combined with the instructions in the developer guide should be a good starting point. Let me know if after reading them you still have doubts :)
